core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1619,6 +1620,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Adds a stop hook which can be used to clean up additional resource. This is called when the
+   * sparkContext is being stopped.
+   */
+  private[spark] def addStopHook(hook: () => Unit): Unit = {
```

Comment:
Does this method (or the collection that it manipulates) need to be synchronized in order to guard against race conditions?